components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/util/NotificationUtil.java

```diff
@@ -556,40 +565,83 @@ public static Notification buildNotification(Event event, Map<String, String> pl
-            organizationName = organizationManager.getOrganizationNameById(associatedOrganizationUUID);
+
+            placeHolderData.put(ORGANIZATION_NAME_PLACEHOLDER,
+                    getOrganizationManager().getOrganizationNameById(tenant.getAssociatedOrganizationUUID()));
+            placeHolderData.put(ORGANIZATION_ID_PLACEHOLDER, tenant.getAssociatedOrganizationUUID());
         } catch (OrganizationManagementClientException e) {
             if (!ERROR_CODE_ORGANIZATION_NOT_FOUND_FOR_TENANT.getCode().equals(e.getErrorCode())) {
                 throw new IdentityEventException(e.getMessage(), e);
             }
         } catch (OrganizationManagementException | UserStoreException e) {
             throw new IdentityEventException(e.getMessage(), e);
         }
-        return organizationName;
+    }
+
+    /**
+     * Fetch the branding preferences from the organization hierarchy by traversing from the base org to the super org.
+     * If branding preferences enabled organization is found, that will be used as the branding preference.
+     *
+     * @param ancestorOrgIds The list of ancestor org ids including the base organization itself.
+     * @param brandingPreferenceManager The branding preference manager.
+     * @return Branding preferences of the immediate organization which has configured branding preferences.
+     */
+    private static BrandingPreference fetchBrandingPreferenceFromOrgHierarchy(List<String> ancestorOrgIds, BrandingPreferenceManager brandingPreferenceManager) {
+
+        String tenantDomain = null;
+        for (String orgId : ancestorOrgIds) {
+            try {
+                tenantDomain = getOrganizationManager().resolveTenantDomain(orgId);
+                int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
+                return brandingPreferenceManager.getBrandingPreference(
+                        BrandingPreferenceMgtConstants.ORGANIZATION_TYPE,
+                        tenantDomain,
+                        BrandingPreferenceMgtConstants.DEFAULT_LOCALE);
+            } catch (BrandingPreferenceMgtException e) {
+                if (!BrandingPreferenceMgtConstants.ErrorMessages.ERROR_CODE_BRANDING_PREFERENCE_NOT_EXISTS.getCode()
+                        .equals(e.getErrorCode())) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Error occurred while retrieving branding preferences for organization " + tenantDomain, e);
+                    }
+                }
+            } catch (OrganizationManagementException e) {
+                log.error("Error occurred while resolving tenant from organization id: " + orgId, e);
```

Comment:
what is the expected default behavior here? this logic implies that if `OrganizationManagementException` occurs then we won't proceed further is that correct or are we handling this at top-level?